### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by shabnamnazmids** [sktime/sktime#5496]
**Describe the bug**

Using ```FeatureUnion()``` to concatenate features extracted from target y, with exogenous features X in a ```ForecatingPipeline()```, the ```.predict()``` function returns an unexpected TypeError, as follows:

TypeError: _transform output of <class 'sktime.transformations.compose._ytox.YtoX'> does not comply with sktime mtype specifications.

**To Reproduce**

```python
from sktime.forecasting.compose import make_reduction
from sktime.regression.deep_learning import CNNRegressor
from sktime.transformations.compose import Id, YtoX
from sktime.transformations.series.lag import Lag
from sktime.transformations.series.impute import Imputer

regressor = CNNRegressor()
forecaster = make_reduction(regressor, window_length=12, strategy="recursive")

pipeline = (Id() + YtoX() * (Lag([1], index_out="original") * Imputer("nearest"))) ** forecaster

pipeline.fit(y=y, X=X_train)
y_pred = pipeline.predict(fh=[1, 2, 3], X=X_test)
```

**Expected behavior**

The expected output would be a series containing the predicted values of y,  consistent with the forecast horizon's length.

**Additional context**

The pipeline functions as expected when using only exogenous features X, or using only features based on y. Detail codes are shared in the [Discussions](https://github.com/sktime/sktime/discussions/5420#discussion-5733845).

**Versions**
0.24.0

</details>

**Comment by fkiraly** [comment]
I can reproduce the problem, it seems to do with whether there is `X` or not passed.

Positive example, with data:

```python
from sktime.forecasting.compose import make_reduction
from sklearn.ensemble import HistGradientBoostingRegressor
from sktime.transformations.compose import Id, YtoX
from sktime.transformations.series.lag import Lag
from sktime.transformations.series.impute import Imputer
from sktime.datasets import load_airline
from sktime.split import temporal_train_test_split

y = load_airline()
y_train, y_test = temporal_train_test_split(y)
X_train = y_train
X_test = y_test

regressor = HistGradientBoostingRegressor()
forecaster = make_reduction(regressor, window_length=12, strategy="recursive")

pipeline = (Id() + YtoX() * (Lag([1], index_out="original") * Imputer("nearest"))) ** forecaster

pipeline.fit(y=y, X=X_train)
y_pred = pipeline.predict(fh=[1, 2, 3], X=X_test)
```

### these work

```python
from sktime.forecasting.compose import make_reduction
from sklearn.ensemble import HistGradientBoostingRegressor
from sktime.transformations.compose import Id, YtoX
from sktime.transformations.series.lag import Lag
from sktime.transformations.series.impute import Imputer
from sktime.datasets import load_airline
from sktime.split import temporal_train_test_split

y = load_airline()
y_train, y_test = temporal_train_test_split(y)
X_train = y_train
X_test = y_test

regressor = HistGradientBoostingRegressor()
forecaster = make_reduction(regressor, window_length=12, strategy="recursive")

pipeline = (Id() + YtoX() * (Lag([1], index_out="original") * Imputer("nearest"))) ** forecaster

pipeline.fit(y=y)
y_pred = pipeline.predict(fh=[1, 2, 3])
```

```python
from sktime.datasets import load_airline
from sktime.forecasting.all import ForecastingHorizon
from sktime.split import temporal_train_test_split
from sktime.registry import craft

y = load_airline()
y_train, y_test = temporal_train_test_split(y)
fh = ForecastingHorizon(y_test.index, is_relative=False)

pipeline = craft("(Id() + YtoX() * (Lag([1], index_out='original') * Imputer('nearest'))) ** NaiveForecaster()")

pipeline.fit(y=y_train)
y_pred = pipeline.predict(fh=fh)
```

**Comment by benHeid** [comment]
I agree with @sbuse's statement in #5127 that this issue is different. 

The problem is that in this code, the following pipeline `YtoX() * (Lag([1], index_out='original') * Imputer('nearest'))` is created. 
This pipeline is executed on `y` data only (due to the `**` dunder). Since the `YtoX() * (Lag([1], index_out='original') * Imputer('nearest'))` is a `TransformerPipeline` it passes these `y` data as `X` data (since transformer are usually only considering `X`). Thus, as input to `YtoX().predict` only X data is passed, which causes an error, since `YtoX` is expecting that y is passed.

Based on these observations, I think we should
* Improve the error message of `YtoX` if either no `y` is passed. Probably introduce some checks before. Currently, it is complaining about wrong mtypes based on the fact that `YtoX` is returning None.
* rethink how YtoX should be integrated into sequential pipelines. Right now, I am unsure how this should work and if YtoX is at the correct position within the pipeline. 

PS. Probably, this is easier to implement using the graphical pipeline.  See the following lines of code

<details>

```python
from sktime.forecasting.compose import make_reduction
from sklearn.ensemble import HistGradientBoostingRegressor
from sktime.transformations.compose import Id, YtoX
from sktime.transformations.series.lag import Lag
from sktime.transformations.series.impute import Imputer
from sktime.datasets import load_airline
from sktime.split import temporal_train_test_split
from sktime.pipeline import Pipeline


y = load_airline()
y_train, y_test = temporal_train_test_split(y)
X_train = y_train
X_test = y_test
pipeline = Pipeline()
pipeline = pipeline.add_step(Id(),
                  name="identity",
                  edges={
                      "X": "y"
                  })

pipeline = pipeline.add_step(Lag([1], index_out="original"),
                             name="lag",
                             edges={
                                 "X": "y"
                             })

pipeline = pipeline.add_step(Imputer("nearest"),
                  name="imputer",
                  edges={
                      "X": "lag"
                  })

regressor = HistGradientBoostingRegressor()
forecaster = make_reduction(regressor, window_length=12, strategy="recursive")

pipeline = pipeline.add_step(forecaster,
                  name="forecaster",
                  edges={
                      "X": ["lag", "identity"],
                      "y": "y"
                  })
pipeline.fit(y=y, X=X_train)
y_pred = pipeline.predict(fh=[1, 2, 3], X=X_test, y=y_test)
```

</details>

**Comment by sbuse** [comment]
Checking out the example from @fkiraly with a linear Regressor shows that the code runs without errors but the model is not built correctly. I think the model should have `window_length * n_features` coefficients, which i think should be 12*2=36 since there are 2 features, y and the lagged y.  `pipeline.get_fitted_params()` shows that the model has only 12 coefficients and playing around with more lags or differences shows that the model only scales with  `window_length`. It looks like the extra features are not used in the model creation.

> ### these work
> ```python
> from sktime.forecasting.compose import make_reduction
> from sklearn.ensemble import HistGradientBoostingRegressor
> from sktime.transformations.compose import Id, YtoX
> from sktime.transformations.series.lag import Lag
> from sktime.transformations.series.impute import Imputer
> from sktime.datasets import load_airline
> from sktime.split import temporal_train_test_split
> from sklearn.linear_model   import Ridge
> 
> y = load_airline()
> y_train, y_test = temporal_train_test_split(y)
> X_train = y_train
> X_test = y_test
> 
> forecaster = make_reduction(Ridge(), window_length=12, strategy="recursive")
> 
> pipeline = (Id() + YtoX() * (Lag([1], index_out="original") * Imputer("nearest"))) ** forecaster
> 
> pipeline.fit(y=y)
> y_pred = pipeline.predict(fh=[1, 2, 3])
> 
> pipeline.get_fitted_params()
> ```

I also agree to with @benHeid on YtoX and the difficulty using it. I still feel like i'm using it the wrong way and it certainly happend that i did so.

**Comment by fkiraly** [comment]
>  shows that the code runs without errors but the model is not built correctly. 

But that would be the fault of the recursive reducer, which as we now know (per bug as described in https://github.com/sktime/sktime/issues/5127) ignores `X`, not of the `YtoX` transformer, no?

**Comment by sbuse** [comment]
I think quite the opposite is the case. For https://github.com/sktime/sktime/issues/5127 I checked that the model is built correctly and passing a different `X` leads to a different result, so everything is fine with the model. @benHeid verified that passing a different `X` leads to different results in https://github.com/sktime/sktime/issues/5127.  The problem there was the handling of the case when `X` is too short to do a prediction and no error is raised. 

Here the model is not built correctly meaning the extra `X` are not used in the model creation.

**Comment by fkiraly** [comment]
well, the workaround is possibly the graph pipeline, see here: https://github.com/sktime/sktime/pull/5509

I will look into this.
